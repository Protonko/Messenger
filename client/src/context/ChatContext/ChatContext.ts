import {createContext} from 'react'

export interface IChatContext {
  selectedMessagesIds: string[]
  toggleSelectMessageId: (selectedMessageId: string) => void
  resetSelectedMessagesIds: () => void
}

export const defaultChatContextValue: IChatContext = {
  selectedMessagesIds: [],
  toggleSelectMessageId: () => {},
  resetSelectedMessagesIds: () => {},
}

export const ChatContext = createContext<IChatContext>(defaultChatContextValue)
