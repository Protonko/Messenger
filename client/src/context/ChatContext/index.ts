export * from './ChatContext'
export * from './ChatContextProvider'
