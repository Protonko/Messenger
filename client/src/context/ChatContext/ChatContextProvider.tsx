import {FC, useState} from 'react'
import {ChatContext, defaultChatContextValue} from 'context/ChatContext'

export const ChatContextProvider: FC = ({children}) => {
  const [selectedMessagesIds, setSelectedMessagesIds] = useState(
    defaultChatContextValue.selectedMessagesIds,
  )

  const resetSelectedMessagesIds = () => {
    setSelectedMessagesIds([])
  }

  const toggleSelectMessageId = (messageId: string) => {
    if (selectedMessagesIds.includes(messageId)) {
      setSelectedMessagesIds((prevState) =>
        prevState.filter((id) => id !== messageId),
      )
    } else {
      setSelectedMessagesIds([...selectedMessagesIds, messageId])
    }
  }

  return (
    <ChatContext.Provider
      value={{
        selectedMessagesIds,
        toggleSelectMessageId,
        resetSelectedMessagesIds,
      }}
    >
      {children}
    </ChatContext.Provider>
  )
}
