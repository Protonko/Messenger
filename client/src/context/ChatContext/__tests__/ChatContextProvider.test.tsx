import {mount} from 'enzyme'
import {ChatContextProvider} from 'context/ChatContext/ChatContextProvider'

describe('ChatContextProvider', () => {
  const children = <div>123</div>
  it('Should render component', () => {
    expect(
      mount(<ChatContextProvider>{children}</ChatContextProvider>),
    ).toMatchSnapshot('ChatContextProvider')
  })
})
