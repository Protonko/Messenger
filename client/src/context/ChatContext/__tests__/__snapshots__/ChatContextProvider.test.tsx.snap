// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatContextProvider Should render component: ChatContextProvider 1`] = `
<ChatContextProvider>
  <div>
    123
  </div>
</ChatContextProvider>
`;
